gpg/gpg.go

```diff
@@ -0,0 +1,62 @@
+// Package gpg has procedures for dealing with GNU Privacy Guard
+// (gpg), in service of signing commits.
+package gpg
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+// ImportKeys looks for all keys in a directory, and imports them into
+// the current user's keyring. A path to a directory or a file may be
+// provided. If the path is a directory, regular files in the
+// directory will be imported, but not subdirectories (i.e., no
+// recursion). It returns the basenames of the succesfully imported
+// keys.
+func ImportKeys(src string) ([]string, error) {
+	info, err := os.Stat(src)
+	var files []string
+	switch {
+	case err != nil:
+		return nil, err
+	case info.IsDir():
+		infos, err := ioutil.ReadDir(src)
+		if err != nil {
+			return nil, err
+		}
+		for _, f := range infos {
+			filepath := filepath.Join(src, f.Name())
+			if f, err = os.Stat(filepath); err != nil {
+				continue
+			}
+			if f.Mode().IsRegular() {
+				files = append(files, filepath)
+			}
+		}
+	default:
+		files = []string{src}
+	}
+
+
+	var imported []string
+	for _, path := range files {
+		if err := gpgImport(path); err != nil {
```

Comment:
I don't think we want this to simply fail silently (at least not in all cases)

Reply:
In what cases would we want to inform the user, and almost as important; how detailed do we want to present it to the user?

Reply:
Things like badly formated data, for instance